src/main/java/core/basesyntax/strategy/operation/BuyOperationHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BuyOperationHandler implements OperationHandler {
+    @Override
+    public void action(FruitTransaction fruitTransaction) {
+        reduceIf(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+    }
+
+    private void reduceIf(String name, int quantity) {
+        if (Storage.fruits.containsKey(name) && Storage.fruits.get(name) >= quantity) {
+            Storage.fruits.put(name, Storage.fruits.get(name) - quantity);
```

Comment:
move Storage.fruits.get(name) to local variable

Reply:
I am checking for the existence of the fruit here, so it seems to me that the variable will be redundant. Well, if necessary - let it be with a variable.